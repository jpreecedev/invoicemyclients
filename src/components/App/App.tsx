import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect, RouteProps } from 'react-router-dom';
import firebase from 'firebase/app';
import 'firebase/analytics';
import 'firebase/auth';
import 'firebase/database';
import { useAuthState } from 'react-firebase-hooks/auth';

import { ROUTES } from '../../routes';
import { LandingPage } from '../../pages/Landing';
import { SignUpPage } from '../../pages/SignUp';
import { ErrorPage } from '../../pages/Error';
import { DashboardPage } from '../../pages/Dashboard';
import { InvoicesPage } from '../../pages/Invoices';
import { InvoicesAddPage } from '../../pages/InvoicesUpdate';
import { ClientsPage } from '../../pages/Clients';
import { ClientsUpdatePage } from '../../pages/ClientsUpdate';

const firebaseConfig = {
	apiKey: process.env.REACT_APP_API_KEY,
	authDomain: process.env.REACT_APP_AUTH_DOMAIN,
	databaseURL: process.env.REACT_APP_DATABASE_URL,
	projectId: process.env.REACT_APP_PROJECT_ID,
	storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
	messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
	appId: process.env.REACT_APP_APP_ID,
	measurementId: process.env.REACT_APP_MEASUREMENT_ID,
};

firebase.initializeApp(firebaseConfig);
firebase.analytics();

firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL);

const AuthenticatedRoute: React.FC<RouteProps> = (props) => {
	const [user, initialising] = useAuthState(firebase.auth());
	if (initialising) {
		return null;
	}

	if (user) {
		return <Route {...props} />;
	}
	return <Redirect to={ROUTES.signup} />;
};

const App: React.FC = () => {
	return (
		<Router>
			<Switch>
				<Route exact path={ROUTES.home}>
					<LandingPage />
				</Route>
				<Route exact path={ROUTES.signup}>
					<SignUpPage />
				</Route>
				<AuthenticatedRoute exact path={ROUTES.dashboard}>
					<DashboardPage />
				</AuthenticatedRoute>
				<AuthenticatedRoute exact path={ROUTES.invoicing.base}>
					<InvoicesPage />
				</AuthenticatedRoute>
				<AuthenticatedRoute exact path={`${ROUTES.invoicing.addEdit}/:id?`}>
					<InvoicesAddPage />
				</AuthenticatedRoute>
				<AuthenticatedRoute exact path={ROUTES.clients.base}>
					<ClientsPage />
				</AuthenticatedRoute>
				<AuthenticatedRoute exact path={`${ROUTES.clients.addEdit}/:id?`}>
					<ClientsUpdatePage />
				</AuthenticatedRoute>
				<Route path="*">
					<ErrorPage />
				</Route>
			</Switch>
		</Router>
	);
};

export { App };
