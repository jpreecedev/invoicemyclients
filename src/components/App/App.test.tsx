import React from 'react'
import { render } from '@testing-library/react'
import { App } from './App'

describe('<App /> tests', () => {
  it('should match snapshot', () => {
    expect(render(<App />).container).toMatchSnapshot()
  })
})
