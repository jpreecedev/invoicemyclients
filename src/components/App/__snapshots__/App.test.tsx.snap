// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients/list"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <section
    class="position-relative pt-11 pt-md-14 mt-n11"
  >
    <div
      class="shape shape-fluid-x shape-blur-2 svg-shim text-light"
    >
      <svg
        fill="none"
        viewBox="0 0 1313 768"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M442.794 768C605.882 768 748.362 679.773 825.111 548.444L825.294 549C825.294 549 825.543 548.251 826.056 546.819C839.139 524.212 850.279 500.339 859.248 475.43C901.996 397.81 989.306 277.09 1144.29 206L1143.37 205.307C1230.34 171.296 1295.63 94.049 1312.83 0H1.29419V295.514C0.631105 305.423 0.294189 315.422 0.294189 325.5C0.294189 569.886 198.408 768 442.794 768Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="container"
    >
      <div
        class="row align-items-center text-center text-md-left"
      >
        <div
          class="col-12 col-md-6"
        >
          <img
            alt="InvoiceMyClients.com"
            class="img-fluid float-right float-md-none mb-6 mb-md-0"
            src="illustration-4.png"
          />
        </div>
        <div
          class="col-12 col-md-6"
        >
          <h1
            class="display-1 font-weight-bold"
          >
            <div>
              Get paid. Fast.
            </div>
          </h1>
          <h2>
            We make it quick, easy, and secure for your clients to pay their invoices.
          </h2>
          <a
            class="btn btn-primary lift mt-3"
            href="/sign-up"
          >
            It's free to get started 
            <svg
              class="ml-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
              <polyline
                points="12 5 19 12 12 19"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </section>
  <section
    class="pt-8 pt-md-11"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-12 col-md-10 col-lg-8 text-center"
        >
          <h6
            class="text-uppercase text-primary font-weight-bold"
          >
            Complete visibility
          </h6>
          <h1
            class="font-weight-bold"
          >
            See your figures at a glance.
          </h1>
          <p
            class="font-size-lg text-muted mb-7"
          >
            Via our dashboard, we make it easy for you to track your invoices, income and revenue over time.
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12 order-md-2"
        >
          <div
            class="device-combo device-combo-macbook-iphonex mb-6 mb-md-0"
          >
            <div
              class="device device-macbook"
            >
              <img
                alt="..."
                class="device-screen"
                src="dashkit-alt.jpg"
              />
              <img
                alt="..."
                class="img-fluid"
                src="macbook.svg"
              />
            </div>
            <div
              class="device device-iphonex"
            >
              <img
                alt="..."
                class="device-screen"
                src="dashkit-alt.jpg"
              />
              <img
                alt="..."
                class="img-fluid"
                src="iphonex.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="py-8 bg-gray-200 mt-5 pt-9 pt-md-11 mt-8 mt-md-11"
  >
    <h2
      class="text-center font-weight-bold mb-8"
    >
      InvoiceMyClients.com, where speed and security matters.
    </h2>
    <div
      class="container"
    >
      <div
        class="row mb-5"
      >
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="18"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="3"
              />
              <rect
                height="9"
                width="3"
                x="7"
                y="7"
              />
              <rect
                height="5"
                width="3"
                x="14"
                y="7"
              />
            </svg>
          </div>
          <h3>
            Invoicing Made Easy
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            We provide an intuitive, simple to use invoice builder with attractive designs
          </p>
        </div>
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="12"
              />
              <line
                x1="12"
                x2="12"
                y1="16"
                y2="16"
              />
            </svg>
          </div>
          <h3>
            Nagging
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            We politely but firmly remind your client that their invoice is due, or overdue.
          </p>
        </div>
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
          </div>
          <h3>
            Secure
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            We use a secure third-party payment processor to ensure the best security and legal compliance.
          </p>
        </div>
      </div>
      <div
        class="row mb-5"
      >
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="1"
                y2="23"
              />
              <path
                d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
              />
            </svg>
          </div>
          <h3>
            Track Your Revenue
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            Use our attractive charts to easily track your revenue and income over time.
          </p>
        </div>
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2"
              />
              <rect
                height="4"
                rx="1"
                ry="1"
                width="8"
                x="8"
                y="2"
              />
            </svg>
          </div>
          <h3>
            Full Audit History
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            We keep detailed logs about every edit, update, and payment you receive for easier record keeping
          </p>
        </div>
        <div
          class="col-12 col-md-4 text-center"
        >
          <div
            class="icon text-primary mb-3"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
              />
            </svg>
          </div>
          <h3>
            Fully Customisable
          </h3>
          <p
            class="text-muted mb-6 mb-md-0"
          >
            Don't like our email templates? Never fear, full customisation is here! Tweak to your hearts content.
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    class="py-8 mt-8 mt-md-10 mb-10"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-12 col-md-5"
        >
          <div
            class="card shadow-lg mb-6 mb-md-0"
          >
            <div
              class="card-body"
            >
              <div
                class="text-center mb-3"
              >
                <span
                  class="badge badge-pill badge-primary-soft"
                >
                  <span
                    class="h6 text-uppercase"
                  >
                    Free
                  </span>
                </span>
              </div>
              <div
                class="d-flex justify-content-center"
              >
                <span
                  class="price display-2 mb-0"
                >
                  $0.00
                </span>
              </div>
              <p
                class="text-center text-muted mb-5"
              >
                Free Forever
              </p>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p>
                  Every feature, free for life.
                </p>
              </div>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p
                  class="mb-5"
                >
                  Never pay a penny.
                </p>
              </div>
              <a
                class="btn btn-block btn-primary"
                href="/sign-up"
              >
                Sign Up Now 
                <i
                  class="fe fe-arrow-right ml-3"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-5"
        >
          <div
            class="card shadow-lg mb-md-0"
          >
            <div
              class="card-body"
            >
              <div
                class="text-center mb-3"
              >
                <span
                  class="badge badge-pill badge-primary-soft"
                >
                  <span
                    class="h6 text-uppercase"
                  >
                    Pay As You Get Paid
                  </span>
                </span>
              </div>
              <div
                class="d-flex justify-content-center"
              >
                <span
                  class="price display-2 mb-0"
                  data-annual="49"
                  data-monthly="69"
                >
                  From 2.4%
                </span>
              </div>
              <p
                class="text-center text-muted mb-5"
              >
                Pay when your client pays you
              </p>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p>
                  2.4% for European Payments
                </p>
              </div>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p
                  class="mb-5"
                >
                  3.9% for International Payments
                </p>
              </div>
              <a
                class="btn btn-block btn-primary-soft"
                href="/sign-up"
              >
                Sign Up Now 
                <i
                  class="fe fe-arrow-right ml-3"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="py-10 py-md-13 bg-gray-200 bg-between"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-12 col-md-10 col-lg-8 text-center"
        >
          <h2
            class="font-weight-bold"
          >
            Try InvoiceMyClients.com for free.
          </h2>
          <p
            class="font-size-lg text-muted mb-8 px-lg-9"
          >
            We can help you save time, save money, and help make it easy for your clients to pay your invoices on time.
          </p>
          <a
            class="btn btn-primary lift mt-3"
            href="/sign-up"
          >
            It's free to get started 
            <svg
              class="ml-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
              <polyline
                points="12 5 19 12 12 19"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </section>
  <footer
    class="py-8 py-md-11 bg-dark"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <p
            class="text-white"
          >
            InvoiceMyClients.com
          </p>
          <a
            class="text-gray-700"
            href="https://jpreecedev.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            © Jon Preece Development Ltd.
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
